apps/Learning-Buddy/screens/NewQuizScreen.jsx

```diff
@@ -0,0 +1,354 @@
+import { View, Text, ScrollView, StyleSheet } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+import React, { useState, useEffect } from 'react';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import {
+  Card,
+  IconButton,
+  Divider,
+  Paragraph,
+  TextInput
+} from 'react-native-paper';
+import BigButton from '../components/BigButton';
+import { colors } from '../config/colors';
+
+export const NewQuizScreen = () => {
+  const insets = useSafeAreaInsets();
+  const navigation = useNavigation();
+
+  const [text, setText] = useState('');
+  const [numLines, changeNumLines] = useState(1);
+
+  useEffect(() => {
+    const charLen = text.length;
+    const suggestLines = Math.ceil(charLen / 50);
+    suggestLines > 0 ? changeNumLines(suggestLines) : changeNumLines(1);
+  }, [text]);
+
+  const [numQuestions, setNumQuestions] = useState(1);
+
+  const handleIncrement = () => {
+    setNumQuestions(numQuestions + 1);
+  };
+
+  const handleDecrement = () => {
+    if (numQuestions > 1) {
+      setNumQuestions(numQuestions - 1);
+    }
+  };
+
+  const [selectedQuestionType, setSelectedQuestionType] = useState(null);
+
+  const handleQuestionTypePress = (questionTypeContent) => {
+    setSelectedQuestionType(questionTypeContent);
+  };
+
+  const [selectedDifficulty, setSelectedDifficulty] = useState(null);
+
+  const handleDifficultyPress = (difficultyContent) => {
+    setSelectedDifficulty(difficultyContent);
+  };
+
+  return (
+    <View
+      style={{
+        // Paddings to handle safe area
+        paddingTop: insets.top,
+        paddingBottom: insets.bottom,
+        paddingLeft: insets.left,
+        paddingRight: insets.right
+      }}
+    >
+      <ScrollView>
+        <View style={{ flex: 1, marginHorizontal: 35 }}>
+          <View>
+            <Text style={localStyles.pageTitle}>Create New Quiz</Text>
+          </View>
+          <Text style={localStyles.title}>Source Material</Text>
+          <View style={localStyles.container}>
+            <TextInput
+              mode='flat'
+              underlineColor={colors.white}
+              activeUnderlineColor={colors.white}
+              style={localStyles.input}
+              label='Enter text'
+              onChangeText={(text) => setText(text)}
+              value={text}
+              multiline
+              numberOfLines={numLines}
+            />
+            {/* <View style={localStyles.buttonContainer}>
+              <BigButton
+                buttonColor={colors.grey}
+                textColor={colors.black}
+                content={'Paste'}
+                onPress={() => {}}
+              />
+            </View> */}
+          </View>
+          <View style={localStyles.divider}>
+            <Divider />
+          </View>
+          <View>
```

Comment:
In the product board, and related to the sprint's story itself, David and I have been talking about the number of questions that can be generated (multiples of 5, or based on the character length of text inputted, etc.) Currently I can ask for any number of questions (I have 999 so far without issue); also, when it goes below 0 it says "NaN" and I can't change it to a number. For now, you can limit it to 10 questions and that should be fine, until David makes a definitive decision about what he wants. This coul
![Screenshot_1688873790](https://github.com/new-id
![Screenshot_1688873814](https://github.com/new-idea-machine/os-lesson-delivery-system/assets/105380332/3f62db22-37cd-453f-b811-1bf2cd490f2d)
ea-machine/os-lesson-delivery-system/assets/105380332/e68d7aa8-17b0-46e1-84a7-c8573e020583)
d mean that if the number is greater than 10, the increment button becomes disabled, or that changing it to a number greater than 10 force-resets it to 10.

Reply:
I've limited the number of questions to six.
I've talked with Muke about this, and I believe it might be preferable to disable user input and only permit changing the value using the arrow buttons since we don't intend to generate many questions.